uiowa/uiowa
# Use Vagrant rsync mounted drives for frontend assets.

Kernel panics are logging `BSD process name corresponding to current thread: nfsd` which makes me think the large number of files synced over NFS between the host and VM is causing crashes. I've experienced crashes mostly running `yarn` commands but have seen it with when viewing multiple sites in my browser and with other CLI commands as well.

DrupalVM documents how to mix synced folder types to improve performance: http://docs.drupalvm.com/en/latest/other/performance/#mixing-synced-folder-types This PR implements that to stop syncing certain directories, most notably `node_modules` which means frontend build tasks need to run on your host. All workspace dependencies are hoisted to the root of the application to avoid node_modules in subdirectories.

Unfortunately, this change probably requires removing `vm.enable` from your `local.blt.yml` file to avoid blt suggesting you run frontend on the VM. This makes it more confusing where to run certain commands.


# How to test
### Testing
- Remove `vm.enable` from `local.blt.yml` file if set.
- Remove all node_module directories:
  - `rm -rf **/node_modules`
- Fresh VM:
  - `vagrant destroy`
  - `vagrant up`
- Follow README docs on frontend requirements installation.
  - `nvm use`
  - `npm install -g yarn`
- Build the frontend.
  - `blt frontend`
- Watch the uids_base workspace and make file changes.
  - `yarn workspace uids_base watch`
- Watch the uiowa_alerts workspace and make file changes.
  - `yarn workspace uiowa_alerts watch`
  - ES6 file changes should work as well. They did not previously on VM.
- Sync the following sites and check the themes:
  - default
  - hr.uiowa.edu
  - uiowa.edu
  - admissions.uiowa.edu
  - ighn.international.uiowa.edu
- Change code files in your editor and make sure they are synced to the VM.
